src/sourcegraph.d.ts

```diff
@@ -717,6 +717,14 @@ declare module 'sourcegraph' {
         ): ProviderResult<Location[]>
     }
 
+    export interface QueryTransformerProvider {
```

Comment:
Add tsdoc docstrings to:

- interface QueryTransformerProvider
- QueryTransformerProvider#transformQuery
- namespace search
- function registerQueryTransformer